`rdkit-stubs/Chem/rdchem.pyi`: `Mol.GetAtoms` and `Mol.GetBonds` are wrong

**Describe the bug**


`lib/python3.12/site-packages/rdkit-stubs/Chem/rdchem.pyi`:

```python
class Mol(Boost.Python.instance):
    # omitted
    @staticmethod
    def GetAtoms(m):
        ...
    @staticmethod
    def GetBonds(m):
        ...
```

Here the signatures of `Mol.GetAtoms` and `GetBonds` are wrong.

**To Reproduce**

```sh
python3 -m venv venv
. ./venv/bin/activate
pip install rdkit==2023.9.6
```

**Expected behavior**

```python
    # not a staticmethod
    def GetAtoms(self) -> rdkit.Chem._GetAtomsIterator:
        ...
    def GetBonds(self) -> rdkit.Chem._GetBondsIterator:
        ...
```

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Configuration (please complete the following information):**
 - RDKit version: 2023.9.6
 - OS: EndeavourOS
 - Python version (if relevant):
 - Are you using conda? no
 - If you are using conda, which channel did you install the rdkit from?
 - If you are not using conda: how did you install the RDKit? pip

**Additional context**
Add any other context about the problem here.

## Describe the bug:

I am seeing the same bug and wanted to provide additional context in case it is helpful.  In my case, I am using Python 3.11, and performing type checking with `mypy` (Version 1.10.1) picks up on these unexpected signatures.

Current version of `lib/python3.11/site-packages/rdkit-stubs/Chem/rdchem.pyi`:
```
class Mol(Boost.Python.instance):
    # omitted
    @staticmethod
    def GetAtoms(m):
        ...
    @staticmethod
    def GetBonds(m):
        ...
```

Minimal example of `__init__.py`:
```
from pydantic import validate_call, ConfigDict

config = ConfigDict(strict=True)

@validate_call(config=config)
def myfunc(smiles: str):
    mol = Chem.MolFromSmiles(smiles)
    for atom in mol.GetAtoms():
        ...
```

Results of type checking the `__init__.py` file above using the existing `lib/python3.11/site-packages/rdkit-stubs/Chem/rdchem.pyi` file:
```
❯ mypy . --check-untyped-defs
... error: Missing positional argument "m" in call to "GetAtoms" of "Mol" [call-arg]
```

I updated the `rdchem.pyi` file locally, analogously to e-kwsm, by removing the `@staticmethod` decorators and adding `self` as an argument:
```
class Mol(Boost.Python.instance):
    # omitted
    def GetAtoms(self) -> rdkit.Chem._GetAtomsIterator:
        ...
    def GetBonds(self) -> rdkit.Chem._GetBondsIterator:
        ...
```

With that change made, the `mypy` type checks pass.
```
❯ mypy . --check-untyped-defs
Success: no issues found in 2 source files
```

I stopped shy of submitting a PR because I was unsure of potential downstream consequences.  Anyway, I hope some of the information above is helpful.

## Configuration (please complete the following information):
RDKit version: `2023.9.6`
OS: `Mac OS`
Python version (if relevant): `python3.11`
Are you using conda? No
If you are using conda, which channel did you install the rdkit from? N/A
If you are not using conda: how did you install the RDKit? `pip`

--- a/rdkit/Chem/__init__.py
+++ b/rdkit/Chem/__init__.py
@@ -100,8 +100,8 @@ class _GetBondsIterator(_GetRDKitObjIterator):
     return self._mol.GetBondWithIdx(i)
 
 
-rdchem.Mol.GetAtoms = lambda m: _GetAtomsIterator(m)
-rdchem.Mol.GetBonds = lambda m: _GetBondsIterator(m)
+rdchem.Mol.GetAtoms = lambda self: _GetAtomsIterator(self)
+rdchem.Mol.GetBonds = lambda self: _GetBondsIterator(self)
 
 
 def QuickSmartsMatch(smi, sma, unique=True, display=False):

